panther_manager/include/panther_manager/manager_bt_node.hpp

```diff
@@ -9,30 +10,41 @@
 #include <ros/ros.h>
 
 #include <std_msgs/Bool.h>
+#include <sensor_msgs/BatteryState.h>
 
+#include <panther_msgs/DriverState.h>
 #include <panther_msgs/IOState.h>
+#include <panther_msgs/SystemStatus.h>
+
+#include <panther_manager/moving_average.hpp>
 
 namespace panther_manager
 {
 
 class ManagerNode
 {
 public:
-  ManagerNode(std::shared_ptr<ros::NodeHandle> nh);
+  ManagerNode(std::shared_ptr<ros::NodeHandle> nh, std::shared_ptr<ros::NodeHandle> pnh);
   ~ManagerNode() {}
 
 private:
+  float battery_percent_;
+  float update_charging_anim_step_;
+  float shutdown_timeout_;
   std::string node_name_;
-  std::string xml_filename_;
+  std::optional<unsigned> battery_status_;
   std::optional<bool> e_stop_state_;
   std::optional<panther_msgs::IOState::ConstPtr> io_state_;
-  std::vector<std::string> plugin_libs_;
 
+  ros::Subscriber battery_sub_;
+  ros::Subscriber driver_state_sub_;
   ros::Subscriber e_stop_sub_;
   ros::Subscriber io_state_sub_;
+  ros::Subscriber system_status_sub_;
   ros::Timer lights_tree_timer_;
   ros::Timer safety_tree_timer_;
   std::shared_ptr<ros::NodeHandle> nh_;
+  std::shared_ptr<ros::NodeHandle> pnh_;
```

Comment:
```suggestion
  std::shared_ptr<ros::NodeHandle> ph_;
```
We discussed this already

Reply:
Changed everywhere. Resolving this